### PR TITLE
[Maintenance] Convert ExpenseGroup, ExpenseCategory, IncomeCategory components to Typescript.

### PR DESCRIPTION
**Pull request by MikesGlitch**
Convert ExpenseGroup, ExpenseCategory, IncomeCategory components to Typescript.

https://github.com/actualbudget/actual/issues/1483

### CODE REVIEW

File: packages/desktop-client/src/components/budget/SidebarCategory.tsx [hunk 1]
```diff
@@ -36,7 +34,6 @@ function SidebarCategory({
   editing,
   style,
   isLast,
-  onDragChange,
```
Comment (MikesGlitch) [inline packages/desktop-client/src/components/budget/SidebarCategory.tsx:39]: Unused

File: packages/desktop-client/src/components/budget/ExpenseCategory.tsx [hunk 2]
```diff
@@ -1,16 +1,39 @@
...
+  onDelete?: ComponentProps<typeof SidebarCategory>['onDelete'];
+  onDragChange: OnDragChangeCallback<CategoryEntity>;
+  onBudgetAction: (idx: number, action: string, arg: unknown) => void;
+  onShowActivity: (name: string, id: string, idx: number) => void;
+  onReorder: OnDropCallback;
+};
+
 function ExpenseCategory({
   cat,
-  budgetArray,
```
Comment (MikesGlitch) [inline packages/desktop-client/src/components/budget/ExpenseCategory.tsx:13]: unused

File: packages/desktop-client/src/components/budget/ExpenseCategory.tsx [hunk 3]
```diff
@@ -67,7 +90,6 @@ function ExpenseCategory({
           onEditName={onEditName}
           onSave={onSave}
           onDelete={onDelete}
-          onDragChange={onDragChange}
```
Comment (MikesGlitch) [inline packages/desktop-client/src/components/budget/ExpenseCategory.tsx:70]: unused

File: packages/desktop-client/src/components/sort.tsx [hunk 4]
```diff
@@ -51,8 +55,8 @@ export function useDraggable({
     },
     collect: monitor => ({ isDragging: monitor.isDragging() }),
 
-    end(item) {
-      _onDragChange.current({ state: 'end', type, item });
+    end(dragState) {
+      _onDragChange.current({ state: 'end', type, item: dragState.item });
```
Comment (MikesGlitch) [inline packages/desktop-client/src/components/sort.tsx:59]: I've changed this - it was a bug - the type checking picked it up. 

The end function passes in the drag state as its prop. We were passing the full thing through as the "item" which should only contain the item that is being dragged. 

I have checked and this actually isn't getting used anywhere - so the bug wasn't visible anywhere.

**Review by MatissJanis — APPROVED** [review]
Thanks for all this hard work!

**Pull request merged**